CMakeLists.txt

```diff
@@ -6,7 +6,9 @@ find_package(ament_cmake REQUIRED)
 
 macro(build_libyaml)
   set(extra_cmake_args)
-  if(DEFINED CMAKE_BUILD_TYPE)
+
+  get_property(multi_config GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
```

Comment:
This property doesn't exist in CMake 3.5. which is the stated minimum version.

Reply:
Bumped minimum to 3.9 in 62808c0 which is the minimum to get [`GENERATOR_IS_MULTI_CONFIG`](https://cmake.org/cmake/help/v3.9/prop_gbl/GENERATOR_IS_MULTI_CONFIG.html). It's less than ROS Eloquent's minimum CMake version of 3.10.2, so it should be fine to require it on this branch for Foxy.